x-pack/solutions/observability/plugins/apm/server/lib/connections/get_connection_stats/get_stats.ts

```diff
@@ -118,14 +108,35 @@ async function getConnectionStats({
   endWithOffset,
   filter,
   numBuckets,
+  withTimeseries,
 }: {
   apmEventClient: APMEventClient;
   startWithOffset: number;
   endWithOffset: number;
   filter: QueryDslQueryContainer[];
   numBuckets: number;
   after?: { serviceName: string | number; dependencyName: string | number };
+  withTimeseries: boolean;
+  dependencyNames?: string[];
 }) {
+  const statsAggs = {
+    total_latency_sum: {
+      sum: {
+        field: SPAN_DESTINATION_SERVICE_RESPONSE_TIME_SUM,
+      },
+    },
+    total_latency_count: {
+      sum: {
+        field: SPAN_DESTINATION_SERVICE_RESPONSE_TIME_COUNT,
+      },
+    },
+    error_count: {
+      filter: {
+        bool: { filter: [{ terms: { [EVENT_OUTCOME]: [EventOutcome.failure] } }] },
```

Comment:
Q: Is this correct? don't we need the other event outcome count? Why aren't we summing up the `SPAN_DESTINATION_SERVICE_RESPONSE_TIME_COUNT` field?

Reply:
We use the total doc count to calculate the error rate, that's why we are only querying for the event.outcome equals to failure. [Here](https://github.com/cauemarcondes/kibana/blob/dependencies-api-breakdown/x-pack/solutions/observability/plugins/apm/server/lib/connections/get_connection_stats/index.ts#L153)